src/background/stores/assessment-card-selection-store.ts

```diff
@@ -50,12 +69,55 @@ export class AssessmentCardSelectionStore extends PersistentStore<AssessmentCard
         this.assessmentCardSelectionActions.navigateToNewCardsView.addListener(
             this.onNavigateToNewCardsView,
         );
+        this.assessmentActions.scanCompleted.addListener(this.onScanCompleted);
+        this.assessmentActions.resetData.addListener(this.onResetData);
+        this.assessmentActions.resetAllAssessmentsData.addListener(this.onResetAllAssessmentsData);
+        this.assessmentActions.loadAssessment.addListener(this.onLoadAssessment);
+        this.assessmentActions.loadAssessmentFromTransfer.addListener(
+            this.onLoadAssessmentFromTransfer,
+        );
     }
 
     public getDefaultState(): AssessmentCardSelectionStoreData {
-        const defaultValue: AssessmentCardSelectionStoreData = {};
+        this.persistedStateFromAssessmentStore =
+            this.convertAllAssessmentResultsToCardSelectionStoreData(this.assessmentStoreData);
+
+        return this.persistedStateFromAssessmentStore ?? this.persistedState ?? {};
+    }
+
+    private convertAllAssessmentResultsToCardSelectionStoreData(
+        assessmentStoreData: AssessmentStoreData,
+    ) {
+        if (
+            !assessmentStoreData ||
+            !assessmentStoreData.assessments ||
+            isEmpty(assessmentStoreData.assessments)
+        ) {
+            return null;
+        }
 
-        return defaultValue;
+        const assessmentCardSelectionStoreData: AssessmentCardSelectionStoreData = {};
+
+        forOwn(assessmentStoreData.assessments, (assessment, key) => {
+            const cardSelectionStoreData: CardSelectionStoreData = assessmentCardSelectionStoreData[
+                key
+            ] ?? {
```

Comment:
nit: it probably makes sense to extract this to a private `getDefaultTestSelectionData` method.